File: CS/CustomParameterEditorsWPF/Reports/XtraReport1.cs
using System.Collections.Generic;
using DevExpress.XtraReports.UI;
// ...

namespace CustomParameterEditorsWPF.Reports {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
            catNames.Type = typeof(List<object>);
            catNames.Value = new List<object>() { "Confections", "Produce" };
        }

        private void XtraReport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            string filter = string.Empty;
            List<object> categoryNames = (List<object>)catNames.Value;
            for (int i = 0; i < categoryNames.Count; i++) {
                filter += string.Format("[CategoryName] = '{0}'", categoryNames[i]);
                if (i != categoryNames.Count - 1)
                    filter += " OR";
            }
            ((XtraReport)sender).FilterString = filter;
        }

    }
}

File: VB/CustomParameterEditorsWPF/Reports/XtraReport1.vb
Imports System.Collections.Generic
Imports DevExpress.XtraReports.UI

' ...
Namespace CustomParameterEditorsWPF.Reports

    Public Partial Class XtraReport1
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
            catNames.Type = GetType(List(Of Object))
            catNames.Value = New List(Of Object)() From {"Confections", "Produce"}
        End Sub

        Private Sub XtraReport1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs)
            Dim filter As String = String.Empty
            Dim categoryNames As List(Of Object) = CType(catNames.Value, List(Of Object))
            For i As Integer = 0 To categoryNames.Count - 1
                filter += String.Format("[CategoryName] = '{0}'", categoryNames(i))
                If i <> categoryNames.Count - 1 Then filter += " OR"
            Next

            CType(sender, XtraReport).FilterString = filter
        End Sub
    End Class
End Namespace
